### PR TITLE
Instructions store

### PR DESCRIPTION
**Pull request by vmarkushin**
Signed-off-by: Vladislav Markushin <negigic@gmail.com>

### Description of the Change
Functionality for storing and sharing constructed instructions.

### CODE REVIEW

File: iroha/src/account.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ pub struct Account {
     pub id: Id,
     /// Asset's in this `Account`.
     pub assets: BTreeMap<<Asset as Identifiable>::Id, Asset>,
+    /// Associated instruction.
+    pub instruction: Option<Box<Instruction>>,
```
Comment (humb1t) [inline iroha/src/account.rs:19]: You need to put instructions into `assets` as an Asset with `instruction@global` Asset Definition.

File: iroha/src/account.rs [hunk 2]
```diff
@@ -132,6 +136,10 @@ pub mod isi {
         ),
         /// Variant of the generic `Add` instruction for `PublicKey` --> `Account`.
         AddSignatory(<Account as Identifiable>::Id, PublicKey),
+        /// Variant of the generic `Add` instruction for `Instruction` --> `Account`.
+        AddInstruction(<Account as Identifiable>::Id, Box<Instruction>),
+        /// Variant of Iroha Special Instructions executes an ISI associated with the account.
+        ExecuteInstruction(<Account as Identifiable>::Id),
```
Comment (humb1t) [inline iroha/src/account.rs:142]: Then both - AddInstruction and ExecuteInstruction will be a composition or alias of out-of-the-box MintAsset instruction.

File: iroha/src/account.rs [hunk 3]
```diff
@@ -258,6 +280,21 @@ pub mod isi {
...
+            world_state_view: &mut WorldStateView,
+        ) -> Result<(), String> {
+            // TODO: permissions
+            let account = world_state_view
+                .account(&self.destination_id)
+                .ok_or("Failed to find account.")?;
+            account.instruction = Some(self.object.clone());
+            Ok(())
+        }
+    }
```
Comment (humb1t) [inline iroha/src/account.rs:297]: And this block may be added to the `bridge.rs` or `dex.rs` module.

File: iroha/src/account.rs (hunk 2 above)
Comment (vmarkushin) [reply iroha/src/account.rs:142]: @humb1t you mean not to add these as separate ISIs?

**Comment by vmarkushin** [comment]
Closed for now because of irrelevance.

**Pull request closed without merging**